precompile/contract/interfaces.go

```diff
@@ -19,6 +20,29 @@ type StatefulPrecompiledContract interface {
 	Run(accessibleState AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error)
 }
 
+// PredicateContext provides context to stateful precompile predicates
+type PredicateContext struct {
+	SnowCtx            *snow.Context
+	ProposerVMBlockCtx *block.Context
+}
+
+// Predicater is an optional interface for StatefulPrecompiledContracts to implement.
+// If implemented, the predicate will be enforced on every transaction in a block, prior to the block's execution.
+// If VerifyPredicate returns an error, the block will fail verification with no further processing.
+// WARNING: this is not intended to be used for custom precompiles. Backwards compatibility with custom precompiles that
```

Comment:
could we clarify what "Custom precompiles" means in this context?

Reply:
What would you suggest? We've always called precompiles outside of Subnet-EVM, so not sure what else we should call them to clarify here.

Reply:
```
 // WARNING: If you are implementing a custom precompile, beware that subnet-evm
 // will not maintain backwards compatibility of this interface and your code should not
// rely on this. Designed for use only by precompiles that are shipped as part of subnet-evm.
```

Reply:
That's way better, thanks for the suggestion!